### PR TITLE
Bring up-to-date with latest matrix-sdk (5d46b35d)

### PR DESCRIPTION
**Pull request by andybalaam**
In my quest to make method names consistent I first needed to reflect previous changes from matrix-sdk.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # UNRELEASED
 
+-   Update matrix-rust-sdk to `#5d46b35d`, which includes:
+
+    -   Rename `VerificationLevel::PreviouslyVerified` to `VerificationViolation`
+        ([#4067](https://github.com/matrix-org/matrix-rust-sdk/pull/4067)
+
+    -   Rename `UserIdentity` to `OtherUserIdentity` and
+        Rename `UserIdentities` to `UserIdentity`
+        ([#4036](https://github.com/matrix-org/matrix-rust-sdk/pull/4036))
```
Comment (richvdh) [inline CHANGELOG.md:10]: If these are changes that affect the js-side API (I think they are?), can we instead express them in terms of breaking changes to the JS objects (which don't have `::` in their names, among other things)?

**Review by richvdh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (andybalaam) [reply CHANGELOG.md:6]: Thanks, remove references to non-exported things in 220d6a2

File: src/error.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ pub enum DecryptionErrorCode {
     /// The sender device is not cross-signed.
     UnsignedSenderDevice,
     /// The sender's identity is unverified, but was previously verified.
-    SenderIdentityPreviouslyVerified,
+    SenderIdentityVerificationViolation,
```
Comment (richvdh) [inline src/error.rs:25]: this looks like a change we need to mention in the changelog

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # UNRELEASED
 
+-   Update matrix-rust-sdk to `#5d46b35d`, which includes:
+
+    -   Rename `UserIdentity` to `OtherUserIdentity` and
+        ([#4036](https://github.com/matrix-org/matrix-rust-sdk/pull/4036))
```
Comment (richvdh) [inline CHANGELOG.md:6]: and ... ?

To be honest, this feels like it buries the lede a bit. Setting this out as a change to matrix-rust-sdk makes it feel like an internal change that a user of the library doesn't need to worry about. This is a breaking change to the *js* API; the fact it's being done to stay in sync with the rust sdk feels barely here or there. Let's set it out as a breaking change in the same way we have with previous breaking changes.

**Review by richvdh — CHANGES_REQUESTED** [review]

File: src/error.rs (hunk 2 above)
Comment (andybalaam) [reply src/error.rs:25]: Thanks, fixed in f147076

File: CHANGELOG.md (hunk 3 above)
Comment (andybalaam) [reply CHANGELOG.md]: Ah, I didn't realise this was how we were laying it out. Fixed in f147076

**Review by richvdh — APPROVED** [review]
There is a reference to `UserIdentity` in the documentation for `OlmMachine.get_identity` which will need an update.

LGTM otherwise.

**Comment by richvdh** [comment]
(if you converted that reference to be a proper `{@link ...}`, like some of the other docs, then this sort of nonsense would be caught by CI)

**Pull request merged**